share-through/share-through-htb.js

```diff
@@ -0,0 +1,425 @@
+var Classify = require('classify.js');
+var Constants = require('constants.js');
+var Partner = require('partner.js');
+var Size = require('size.js');
+var SpaceCamp = require('space-camp.js');
+var System = require('system.js');
+var Network = require('network.js');
+var Utilities = require('utilities.js');
+var ComplianceService;
+var EventsService;
+var RenderService;
+
+//? if (DEBUG) {
+var ConfigValidators = require('config-validators.js');
+var PartnerSpecificValidator = require('share-through-htb-validator.js');
+var Scribe = require('scribe.js');
+var Whoopsie = require('whoopsie.js');
+//? }
+
+////////////////////////////////////////////////////////////////////////////////
+// Main ////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+
+/**
+ * Partner module template
+ *
+ * @class
+ */
+function ShareThroughHtb(configs) {
+    /* =====================================
+     * Data
+     * ---------------------------------- */
+
+    /* Private
+     * ---------------------------------- */
+
+    /**
+     * Reference to the partner base class.
+     *
+     * @private {object}
+     */
+    var __baseClass;
+
+    /**
+     * Profile for this partner.
+     *
+     * @private {object}
+     */
+    var __profile;
+
+    /* =====================================
+     * Functions
+     * ---------------------------------- */
+
+    /* Utilities
+     * ---------------------------------- */
+
+    /**
+     * Generates the request URL and query data to the endpoint for the xSlots
+     * in the given returnParcels.
+     *
+     * @param  {object[]} returnParcels
+     *
+     * @return {object}
+     */
+    function __generateRequestObj(returnParcels) {
+        var baseUrl = Browser.getProtocol() + '//btlr.sharethrough.com/header-bid/v1';
+
+        var queryObj = {
+            placement_key: returnParcels[0].xSlotRef.placementKey,
+            bidId: returnParcels[0].requestId,
+            instant_play_capable: __canAutoPlayHTML5Video(),
+            hbSource: "indexExchange",
+            hbVersion: "2.1.2",
+            cbust: System.now()
+        };
+
+        var unifiedID = __getUnifiedID(returnParcels);
+        if (unifiedID) {
+            queryObj.ttduid = unifiedID;
+        }
+
+        var privacyEnabled = ComplianceService.isPrivacyEnabled();
+        var gdprStatus = ComplianceService.gdpr.getConsent();
+        if (privacyEnabled && gdprStatus) {
+            queryObj.consent_required = gdprStatus.applies;
+            queryObj.consent_string = gdprStatus.consentString;
+        }
+
+        return {
+            url: baseUrl,
+            data: queryObj,
+        };
+    }
+
+    /* Helpers
+     * ---------------------------------- */
+
+    function __getUnifiedID(returnParcels) {
+        var uids = []
+        try {
+            uids = returnParcels[0].identityData.AdserverOrgIp.data.uids;
```

Comment:
Since returnParcels[0].identityData.AdserverOrgIp.data could by an empty object so we would suggest updating the code to check if the Object.key(returnParcels[0].identityData.AdserverOrgIp.data).length > 0

Reply:
Feedback addressed. Please see updated code with latest commit below. https://github.com/indexexchange/ix-library-adapters/pull/39/commits/63228a3df8476d7a01986ac20d0b2f6051aa5e28